### PR TITLE
Always requeue explicit peers

### PR DESCRIPTION
**Pull request by mafintosh**

### CODE REVIEW

File: lib/peer-info.js [hunk 1]
```diff
@@ -75,6 +76,7 @@ module.exports = class PeerInfo extends EventEmitter {
   }
 
   _updatePriority () {
+    if (this.explicit && this.attempts > 3) this.attempts = 0
```
Comment (andrewosh) [inline lib/peer-info.js:79]: Doesn't your code in `retry-timer.js` depend on `attempts` being > 3?

Comment (mafintosh) [reply lib/peer-info.js:79]: I might be wrong, but the flow is:

1. connect ()
2. attempts++
3. if attempts > 3 use xl timer
4. xl timer calls requeue
5. if attempts > 3 reset to 0 (ie pretend it's a rediscover)

Comment (mafintosh) [reply lib/peer-info.js:79]: let's call reset instead

Comment (andrewosh) [reply lib/peer-info.js:79]: But if the peer's offline, won't that trigger 3 fast connections that will fail. There's no new information at that point to suggest it should be attempted more than once.

Comment (mafintosh) [reply lib/peer-info.js:79]: yuh was just thinking about that. the optimal way should be to see why it failed and react based on that. i think that's too complicated for this PR but we should improve that in the future.

Comment (andrewosh) [reply lib/peer-info.js]: :+1:

**Review by andrewosh — APPROVED** [review]

**Comment by mafintosh** [comment]
4.1.0 (i guess it should be been a patch but whatevs)

**Pull request merged**